libvast/test/system/partition_index.cpp

```diff
@@ -0,0 +1,183 @@
+    std::vector<uuid> result;
+    if (first < ids.size())
+      for (; first != std::min(last, ids.size()); ++first)
+        result.emplace_back(ids[first]);
+    return result;
+  }
+
+  auto slice(size_t index) const {
+    return slice(index, index + 1);
+  }
+
+  auto query(std::string_view hhmmss) {
+    std::string q = "&time == 1970-01-01+";
+    q += hhmmss;
+    q += ".0";
+    return uut.lookup(unbox(to<expression>(q)));
+  }
+
+  auto empty() const {
+    return slice(ids.size());
+  }
+
+  auto query(std::string_view hhmmss_from, std::string_view hhmmss_to) {
+    std::string q = "&time >= 1970-01-01+";
+    q += hhmmss_from;
+    q += ".0";
+    q += " && &time <= 1970-01-01+";
+    q += hhmmss_to;
+    q += ".0";
+    return sort(uut.lookup(unbox(to<expression>(q))));
+  }
+
+  template <class T>
+  T sort(T xs) {
+    std::sort(xs.begin(), xs.end());
+    return xs;
+  }
+};
+
+} // namespace <anonymous>
+
+FIXTURE_SCOPE(partition_index_tests, fixture)
+
+TEST(uuid lookup) {
+  MESSAGE("generate " << num_partitions << " UUIDs for the partitions");
+  for (size_t i = 0; i < num_partitions; ++i)
+    ids.emplace_back(uuid::random());
+  for (size_t i = 0; i < num_partitions; ++i)
+    for (size_t j = 0; j < num_partitions; ++j)
+      if (i != j && ids[i] == ids[j])
+        FAIL("ID " << i << " and " << j << " are equal!");
+  MESSAGE("generate events and add events to the partition index");
+  std::vector<mock_partition> mock_partitions;
+  for (size_t i = 0; i < num_partitions; ++i) {
+    auto& mp = mock_partitions.emplace_back(ids[i], i);
+    uut.add(mp.events, mp.id);
+  }
+  MESSAGE("verify generated timestamps");
+  {
+    auto& p0 = mock_partitions[0];
+    CHECK_EQUAL(p0.range.from, epoch);
+    CHECK_EQUAL(p0.range.to, epoch + 24s);
+    CHECK_EQUAL(p0.range, unbox(uut[ids[0]]).range);
+    auto& p1 = mock_partitions[1];
+    CHECK_EQUAL(p1.range.from, epoch + 25s);
+    CHECK_EQUAL(p1.range.to, epoch + 49s);
+    CHECK_EQUAL(p1.range, unbox(uut[ids[1]]).range);
+    auto& p2 = mock_partitions[2];
+    CHECK_EQUAL(p2.range.from, epoch + 50s);
+    CHECK_EQUAL(p2.range.to, epoch + 74s);
+    CHECK_EQUAL(p2.range, unbox(uut[ids[2]]).range);
+    auto& p3 = mock_partitions[3];
+    CHECK_EQUAL(p3.range.from, epoch + 75s);
+    CHECK_EQUAL(p3.range.to, epoch + 99s);
+    CHECK_EQUAL(p3.range, unbox(uut[ids[3]]).range);
+  }
+  MESSAGE("check whether point queries return correct slices");
+  CHECK_EQUAL(query("00:00:00"), slice(0));
+  CHECK_EQUAL(query("00:00:24"), slice(0));
+  CHECK_EQUAL(query("00:00:25"), slice(1));
+  CHECK_EQUAL(query("00:00:49"), slice(1));
+  CHECK_EQUAL(query("00:00:50"), slice(2));
+  CHECK_EQUAL(query("00:01:14"), slice(2));
+  CHECK_EQUAL(query("00:01:15"), slice(3));
+  CHECK_EQUAL(query("00:01:39"), slice(3));
+  CHECK_EQUAL(query("00:01:40"), empty());
+  MESSAGE("check whether time-range queries return correct slices");
+  CHECK_EQUAL(query("00:00:01", "00:00:10"), slice(0));
```

Comment:
@mavam This unit test fails, but I don't understand why. Do I do something wrong here or is this a bug in `time_restrictor` (used in `partition_index.cpp` for implementing `partition_index::lookup`).